docs/howto/add_link.md

```diff
@@ -0,0 +1,336 @@
+
+### [Resource link](http://jsonapi.org/format/#document-resource-object-links)
+
+You can use the `link` class method to define the links you need in the resource's primary data:
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :name
+
+  link(:self) { user_path(object.id) }
+  link(:posts) { posts_path(user_id: object.id) }
+end
+
+user = User.new(id: 1, name: "Kelly")
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
+resource.to_json
+```
+
+This will result in:
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "Kelly"
+    },
+    "links": {
+      "self": "/users/1",
+      "posts": "/posts?user_id=1"
+    }
+  }
+}
+```
+
+### [Relationship links](http://jsonapi.org/format/#document-resource-object-relationships)
+
+You can use the `link` method wihtin you relationship block in order to define the links you need in the relationships:
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :id, :name
+  has_many :posts do
+    link(:self) do
+      user_posts_path(object.id) # Rails url helper automatically available
+    end 
+  end
+end
+
+user = User.new(id: 1, name: "Kelly", posts: [])
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
+resource.to_json
+```
+
+Note that `user_posts_path` method is one of Rails' url helpers, which are automatically available within the `link` block if you are using Rails.
+
+The previous snippet would result in:
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "Kelly"
+    },
+    "relationships": {
+      "posts": {
+        "data": [],
+        "links": {
+          "self": "users/1/posts"
+        }
+      }
+    }
+  }
+}
+```
+
+If you only want the `links` member to appear you can set `include_data false` within the relationship block:
+
+```ruby
+class UserSerializer < ActiveModel::Serializer
+  attributes :name
+  has_many :posts do
+    link(:self) do
+      include_data false
+      user_posts_path(object.id) # Rails url helper automatically available
+    end 
+  end
+end
+
+user = User.new(id: 1, name: "Kelly", posts: [ Post.new(id: 1) ])
+resource = ActiveModelSerializers::SerializableResource.new(user, adapter: :json_api)
+resource.to_json
+```
+
+This would result in:
+
+```json
+{
+  "data": {
+    "id": "1",
+    "type": "users",
+    "attributes": {
+      "name": "Kelly"
+    },
+    "relationships": {
+      "posts": {
+        "links": {
+          "self": "users/1/posts"
+        }
+      }
+    }
+  }
+}
+```
+
+## JSON adapter
+
+The `json` is not able to handle links defined by the `link` class method. However, you can either use the `meta`  member or define a `links` attribute on the serializer.
```

Comment:
JSON adapter doesn't respect the `link` macro in resources or relationships, however it does respect 'top-level meta', which can be used to include links.  See the implementation of the https://github.com/rails-api/active_model_serializers/blob/v0.10.3/lib/active_model_serializers/adapter/json_api/link.rb for more details